tools/gcov.py

```diff
@@ -0,0 +1,209 @@
+            "gcov end"
+        ):
+            hex_dump += lines[next_line_index].strip()
+            next_line_index += 1
+
+        if size != len(hex_dump) // 2:
+            print(
+                f"Size mismatch for {filename}: expected {size} bytes, got {len(hex_dump) // 2} bytes"
+            )
+
+        checksum_match = (
+            re.search(r"checksum:\s*(0x[0-9a-fA-F]+)", lines[next_line_index])
+            if next_line_index < len(lines)
+            else None
+        )
+        if not checksum_match:
+            continue
+
+        checksum = int(checksum_match.group(1), 16)
+        calculated_checksum = sum(bytearray.fromhex(hex_dump)) % 65536
+        if calculated_checksum != checksum:
+            print(
+                f"Checksum mismatch for {filename}: expected {checksum}, got {calculated_checksum}"
+            )
+            continue
+
+        with open(filename, "wb") as fp:
+            fp.write(bytes.fromhex(hex_dump))
+            print(f"write {filename} success")
+
+
+def collect_match_endswith(endswith, source_dir, target_dir):
+    print(f"Collect {endswith} files {source_dir} -> {target_dir}")
+
+    if not os.path.exists(target_dir):
+        os.makedirs(target_dir)
+
+    for _root, _, _files in os.walk(source_dir):
+        for _file in _files:
+            if _file.endswith(endswith):
+                source_file = os.path.join(_root, _file)
+                target_file = os.path.join(target_dir, _file)
+                shutil.copy2(source_file, target_file)
+
+
+def arg_parser():
+    parser = argparse.ArgumentParser(
+        description="Code coverage generation tool.", add_help=False
+    )
+    parser.add_argument("-i", "--input", help="Input dump data")
+    parser.add_argument("-t", dest="gcov_tool", help="Path to gcov tool")
+    parser.add_argument("-s", dest="gcno_dir", help="Directory containing gcno files")
+    parser.add_argument("-a", dest="gcda_dir", help="Directory containing gcda files")
+    parser.add_argument("--debug", action="store_true", help="Enable debug mode")
+    parser.add_argument(
+        "-x",
+        dest="only_copy",
+        action="store_true",
+        help="Only copy *.gcno and *.gcda files",
+    )
+    parser.add_argument(
+        "gcov_dir",
+        nargs="?",
+        default=os.getcwd(),
+        help="Directory to store gcov data and report",
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = arg_parser()
+
+    root_dir = os.getcwd()
+    gcov_dir = os.path.abspath(args.gcov_dir)
+    gcno_dir = os.path.abspath(args.gcno_dir) if args.gcno_dir else root_dir
+    gcda_dir = os.path.abspath(args.gcda_dir) if args.gcda_dir else root_dir
+
+    coverage_file = os.path.join(gcov_dir, "coverage.info")
+    result_dir = os.path.join(gcov_dir, "result")
+    gcov_data_dir = os.path.join(gcov_dir, "data")
+
+    if args.debug:
+        debug_file = os.path.join(gcov_dir, "debug.log")
+        sys.stdout = open(debug_file, "w+")
```

Comment:
It's counterintuitive when `--debug` is enabled, nothing outputs to the console but silently goes to a file.
How about remove `--debug` directly.

Reply:
The autest process cannot get the running script log information. If an error occurs, it may not be found and saved in this file.